src/main/java/com/taxi/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,229 @@
+                + "WHERE id = ?;";
+        try (Connection connection = connectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            int updatedRows = preparedStatement.executeUpdate();
+            return updatedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t extract data for Car with id "
+                    + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> carList = new ArrayList<>();
+        String getAllStatement = "SELECT c.id, c.model, c.manufacturer_id, c.isDeleted,"
+                + " m.id, m.name, m.country, m.isDeleted"
+                + " FROM (cars c JOIN cars_drivers cd ON c.id = cd.car_id"
+                + " JOIN manufacturers m on c.manufacturer_id = m.id)"
+                + " WHERE cd.driver_id = ? ";
+        try (Connection connection = connectionUtil.getConnection();
+                 PreparedStatement preparedStatement =
+                         connection.prepareStatement(getAllStatement)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(getCar(resultSet));
+            }
+            return carList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all Cars to the List !"
+                    + carList.toString(), e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        Manufacturer manufacturer = getManufacturer(resultSet);
+        car.setId(resultSet.getObject("id", Long.class));
+        car.setModel(resultSet.getString("model"));
+        car.setDeleted(resultSet.getBoolean("isDeleted"));
+        car.setManufacturer(manufacturer);
+        return car;
+    }
+
+    private Manufacturer getManufacturer(ResultSet resultSet) throws SQLException {
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setId(resultSet.getObject("m.id", Long.class));
+        manufacturer.setName(resultSet.getString("name"));
+        manufacturer.setCountry(resultSet.getString("country"));
+        manufacturer.setDeleted(resultSet.getBoolean("m.isDeleted"));
+        return manufacturer;
+    }
+
+    private Driver getDriver(ResultSet resultSet) throws SQLException {
+        Driver driver = new Driver();
+        driver.setId(resultSet.getObject("id", Long.class));
+        driver.setName(resultSet.getString("name"));
+        driver.setLicenseNumber(resultSet.getString("licence"));
+        driver.setDeleted(resultSet.getBoolean("isDeleted"));
+        return driver;
+    }
+
+    private Set<Driver> updateDriversForCar(Car car, Connection connection) {
```

Comment:
Зачем возвращать сет, который мы просто берем у машины и ничего с ним не делаем?
```suggestion
    private void addDriversForCar(Car car, Connection connection) {
```